### PR TITLE
Adapt to Protocol v11

### LINKED ISSUE #197: Protocol 11 Compatibility
Protocol V11 of Stellar adds a new `manage_buy_offer` operation. Also names of `manage_offer` and `create_passive_offer` have been changed to `manage_sell_offer` and `create_passive_sell_offer`.

References:
* https://github.com/stellar/go/pull/1164
* https://github.com/stellar/stellar-protocol/blob/master/core/cap-0006.md
* [XDR files diff](https://github.com/stellar/stellar-core/compare/v10.3.0...v11.0.0#diff-01cb6b2520b4d5097b53837c1c18c5a3) - this shows all changes between v10.3.0 and v11.0.0 in Stellar Core — you'll need to just look at the changes in `src/xdr` (realistically any file that ends with a `.x` extension).

### PR DESCRIPTION
**Pull request by overcat**
Resolve #197

### CODE REVIEW

File: src/main/java/org/stellar/sdk/CreatePassiveOfferOperation.java [hunk 1]
```diff
@@ -1,135 +1,13 @@
...
-
-import java.math.BigDecimal;
-import java.math.MathContext;
-
-import static com.google.common.base.Preconditions.checkNotNull;
-
 /**
- * Represents <a href="https://www.stellar.org/developers/learn/concepts/list-of-operations.html#create-passive-offer" target="_blank">CreatePassiveOffer</a> operation.
- * @see <a href="https://www.stellar.org/developers/learn/concepts/list-of-operations.html" target="_blank">List of Operations</a>
+ * @deprecated Use {@link CreatePassiveSellOfferOperation}
```
Comment (overcat) [inline src/main/java/org/stellar/sdk/CreatePassiveOfferOperation.java]: Should I deprecate a class like this?

File: src/main/java/org/stellar/sdk/ManageOfferOperation.java [hunk 2]
```diff
@@ -1,164 +1,14 @@
...
-
-import java.math.BigDecimal;
-import java.math.MathContext;
-
-import static com.google.common.base.Preconditions.checkNotNull;
-
 /**
- * Represents <a href="https://www.stellar.org/developers/learn/concepts/list-of-operations.html#manage-offer" target="_blank">ManageOffer</a> operation.
- * @see <a href="https://www.stellar.org/developers/learn/concepts/list-of-operations.html" target="_blank">List of Operations</a>
+ * @deprecated Use {@link ManageSellOfferOperation}
```
Comment (overcat) [inline src/main/java/org/stellar/sdk/ManageOfferOperation.java]: Should I deprecate a class like this?

File: src/main/java/org/stellar/sdk/responses/SubmitTransactionResponse.java [hunk 3]
```diff
@@ -97,15 +100,25 @@ public Long getOfferIdFromResult(int position) {
...
+        OperationResult.OperationResultTr operationResultTr = operationResult.getTr();
+
+        if (operationType == OperationType.MANAGE_SELL_OFFER) {
+            if (operationResultTr.getManageSellOfferResult().getSuccess().getOffer().getOffer() == null) {
+                return null;
+            }
+            return operationResultTr.getManageSellOfferResult().getSuccess().getOffer().getOffer().getOfferID().getInt64();
         }
 
-        if (result.getResult().getResults()[0].getTr().getManageOfferResult().getSuccess().getOffer().getOffer() == null) {
```
Comment (overcat) [inline src/main/java/org/stellar/sdk/responses/SubmitTransactionResponse.java]: I think this is incorrect, it should be `if (result.getResult().getResults()[position].getTr().getManageOfferResult().getSuccess().getOffer().getOffer() == null) {`. If you agree with me, you can cherry-pick 555ffcd8d02f9a950564210435a45710c3163a61 first.

**Comment by overcat** [comment]
I am very new in Java world, if you have any suggestions, feel free to let me know.
Thanks.

**Comment by tomquisel** [comment]
This is amazing @overcat! @nullstyle can you prioritize reviewing this?

File: src/main/java/org/stellar/sdk/CreatePassiveOfferOperation.java (hunk 1 above)
Comment (bartekn) [reply src/main/java/org/stellar/sdk/CreatePassiveOfferOperation.java]: We should deprecate this class but it needs to work until removed. We are still pre 1.0.0 version so we can remove it in the next minor version (assuming these changes will be part of 0.7.0, then deprecated methods can be removed in 0.8.0).

File: src/main/java/org/stellar/sdk/ManageOfferOperation.java (hunk 2 above)
Comment (bartekn) [reply src/main/java/org/stellar/sdk/ManageOfferOperation.java]: See the comment under `CreatePassiveOffer`.

File: src/main/java/org/stellar/sdk/responses/SubmitTransactionResponse.java (hunk 3 above)
Comment (bartekn) [reply src/main/java/org/stellar/sdk/responses/SubmitTransactionResponse.java]: Yes, you are correct. Looks like a bug introduced when merging etc.

File: src/main/java/org/stellar/sdk/CreatePassiveOfferOperation.java (hunk 1 above)
Comment (overcat) [reply src/main/java/org/stellar/sdk/CreatePassiveOfferOperation.java]: I've added them back, but some behavior may change due to the XDR changes.

File: src/main/java/org/stellar/sdk/ManageOfferOperation.java [hunk 4]
```diff
@@ -1,164 +1,164 @@
...
+        amount.setInt64(Operation.toXdrAmount(this.amount));
+        op.setAmount(amount);
+        Price price = Price.fromString(this.price);
+        op.setPrice(price.toXdr());
+        Int64 offerId = new Int64();
+        offerId.setInt64(Long.valueOf(this.offerId));
+        op.setOfferID(offerId);
+
+        org.stellar.sdk.xdr.Operation.OperationBody body = new org.stellar.sdk.xdr.Operation.OperationBody();
+        body.setDiscriminant(OperationType.MANAGE_SELL_OFFER);
```
Comment (overcat) [inline src/main/java/org/stellar/sdk/ManageOfferOperation.java]: OperationType.MANAGE_OFFER no longer exists, and the actual return here is OperationType.MANAGE_SELL_OFFER.

Comment (overcat) [inline src/main/java/org/stellar/sdk/ManageOfferOperation.java]: OperationType.MANAGE_OFFER no longer exists, and the actual return here is OperationType.MANAGE_SELL_OFFER.

File: src/main/java/org/stellar/sdk/CreatePassiveOfferOperation.java [hunk 5]
```diff
@@ -1,135 +1,137 @@
...
+        op.setSelling(selling.toXdr());
+        op.setBuying(buying.toXdr());
+        Int64 amount = new Int64();
+        amount.setInt64(Operation.toXdrAmount(this.amount));
+        op.setAmount(amount);
+        Price price = Price.fromString(this.price);
+        op.setPrice(price.toXdr());
+
+        org.stellar.sdk.xdr.Operation.OperationBody body = new org.stellar.sdk.xdr.Operation.OperationBody();
+        body.setDiscriminant(OperationType.CREATE_PASSIVE_SELL_OFFER);
```
Comment (overcat) [inline src/main/java/org/stellar/sdk/CreatePassiveOfferOperation.java]: OperationType. CREATE_PASSIVE_OFFER no longer exists, and the actual return here is OperationType. CREATE_PASSIVE_SELL_OFFER.

Comment (bartekn) [reply src/main/java/org/stellar/sdk/CreatePassiveOfferOperation.java]: Yes, this is correct. The binary representation is backward compatible.

Comment (overcat) [reply src/main/java/org/stellar/sdk/CreatePassiveOfferOperation.java]: Hi, bartekn, thank you for your guidance.

**Pull request merged**